src/v2i-hub/SpatPlugin/src/signalController.cpp

```diff
@@ -177,22 +164,23 @@ void SignalController::start_signalController()
 
 						IsReceiving = 1;
 						pthread_mutex_lock(&spat_message_mutex);
-
+						auto ntcip1202 = std::make_shared<Ntcip1202>();
+						ntcip1202->setSignalGroupMappingList(_signalGroupMappingJson);
 						//printf("Signal Controller calling ntcip1202 copyBytesIntoNtcip1202");
 						ntcip1202->copyBytesIntoNtcip1202(buf, numbytes);
 
 						//printf("Signal Controller calling ntcip1202 ToJ2735r41SPAT");
 						SPAT *_spat = (SPAT *) calloc(1, sizeof(SPAT));
 						ntcip1202->ToJ2735r41SPAT(_spat, _intersectionName, _intersectionId);
-
+						
 						//printf("Signal Controller calling _spatMessage set_j2735_data\n");
 						//_spatMessage.set_j2735_data(_spat);
 						if (_spatMessage != NULL)
 						{
-							delete _spatMessage;
+							// delete _spatMessage;
```

Comment:
Why did you comment this out.

Reply:
We use the smart pointer, so no need to use delete. I will remove the comment out code once Will confirm it.